cmd/dep/godep_importer_test.go

```diff
@@ -18,6 +18,189 @@ import (
+	testCases := map[string]struct {
+		json                 godepJSON
+		wantConvertErr       bool
+		matchPairedVersion   bool
+		projectRoot          gps.ProjectRoot
+		notWantedProjectRoot *gps.ProjectRoot
+		wantConstraint       string
+		wantRevision         *gps.Revision
+		wantVersion          string
+		wantLockCount        int
+	}{
+		"convert project": {
+			json: godepJSON{
+				Imports: []godepPackage{
+					{
+						ImportPath: "github.com/sdboyer/deptest",
+						// This revision has 2 versions attached to it, v1.0.0 & v0.8.0.
+						Rev:     "ff2948a2ac8f538c4ecd55962e919d1e13e74baf",
+						Comment: "v0.8.0",
+					},
+				},
+			},
+			matchPairedVersion: true,
+			projectRoot:        gps.ProjectRoot("github.com/sdboyer/deptest"),
+			wantConstraint:     "^0.8.0",
+			wantRevision:       revisionPtr(gps.Revision("ff2948a2ac8f538c4ecd55962e919d1e13e74baf")),
+			wantVersion:        "v0.8.0",
+			wantLockCount:      1,
+		},
+		"with semver suffix": {
+			json: godepJSON{
+				Imports: []godepPackage{
+					{
+						ImportPath: "github.com/sdboyer/deptest",
+						Rev:        "ff2948a2ac8f538c4ecd55962e919d1e13e74baf",
+						Comment:    "v1.12.0-12-g2fd980e",
+					},
+				},
+			},
+			projectRoot:        gps.ProjectRoot("github.com/sdboyer/deptest"),
+			matchPairedVersion: false,
+			wantConstraint:     ">=1.12.0, <=12.0.0-g2fd980e",
+			wantLockCount:      1,
+		},
+		"empty comment": {
+			json: godepJSON{
+				Imports: []godepPackage{
+					{
+						ImportPath: "github.com/sdboyer/deptest",
+						// This revision has 2 versions attached to it, v1.0.0 & v0.8.0.
+						Rev: "ff2948a2ac8f538c4ecd55962e919d1e13e74baf",
+					},
+				},
+			},
+			projectRoot:        gps.ProjectRoot("github.com/sdboyer/deptest"),
+			matchPairedVersion: true,
+			wantConstraint:     "^1.0.0",
+			wantRevision:       revisionPtr(gps.Revision("ff2948a2ac8f538c4ecd55962e919d1e13e74baf")),
+			wantVersion:        "v1.0.0",
+			wantLockCount:      1,
+		},
+		"bad input - empty package name": {
+			json: godepJSON{
+				Imports: []godepPackage{{ImportPath: ""}},
+			},
+			wantConvertErr: true,
+		},
+		"bad input - empty revision": {
+			json: godepJSON{
+				Imports: []godepPackage{
+					{
+						ImportPath: "github.com/sdboyer/deptest",
+					},
+				},
+			},
+			wantConvertErr: true,
+		},
+		"sub-packages": {
+			json: godepJSON{
+				Imports: []godepPackage{
+					{
+						ImportPath: "github.com/sdboyer/deptest",
+						// This revision has 2 versions attached to it, v1.0.0 & v0.8.0.
+						Rev: "ff2948a2ac8f538c4ecd55962e919d1e13e74baf",
+					},
+					{
+						ImportPath: "github.com/sdboyer/deptest/foo",
+						// This revision has 2 versions attached to it, v1.0.0 & v0.8.0.
+						Rev: "ff2948a2ac8f538c4ecd55962e919d1e13e74baf",
+					},
+				},
+			},
+			projectRoot:          gps.ProjectRoot("github.com/sdboyer/deptest"),
+			notWantedProjectRoot: projectRootPtr(gps.ProjectRoot("github.com/sdboyer/deptest/foo")),
```

Comment:
Between the check for an `Constraint` entry for `projectRoot` and the check on `wantLockCount`, I would be quite happy to remove `notWantedProjectRoot` entirely. It will effectively test what the importer cared about: consolidating multiple subpackages into a single entry for the root package.

Then we can remove `projectRootPtr` as well. Sound good?